### ISSUE TITLE
Odd folder names

### ISSUE THREAD

**Issue by Sparticuz** [gboudreau/Greyhole#186]
Every once in a while, I get these folders with wierd names showing up. They are empty. These seem to have been created when I ran a fsck yesterday. Any Idea what they are?
![image](https://user-images.githubusercontent.com/161913/30431584-017563a8-992d-11e7-8171-16865e8cc0ae.png)

**Comment by gboudreau** [comment]
Never saw anything that looks like those, no.

Can you find any reference to those names in the greyhole logs?
Maybe grep one of the strings that should be unique enough, like `9YW8W`.

**Comment by Sparticuz** [comment]
Here is one. I'm moving my ripped movies from my Windows 10 PC, to my Arch box over samba. This has happened before, but recently, alot more.
````
Sep 14 09:09:40 DEBUG read_smb_spool: Processing Samba spool...
Sep 14 09:09:40 DEBUG read_smb_spool: Directory created: Movies/P�9w�U
Sep 14 09:09:42 DEBUG read_smb_spool: Found 3 new tasks in spool.
Sep 14 09:09:42 INFO write: Now working on task ID 581864: write Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:09:42 INFO write: File created: Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv - 2.00GB
Sep 14 09:09:42 DEBUG write: Loading metafiles for Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv ...
Sep 14 09:09:42 DEBUG write:   Got 0 metadata files.
Sep 14 09:09:42 DEBUG write: 0 metafiles loaded.
Sep 14 09:09:42 DEBUG write:   Drives with available space: /var/drives/alpha/gh (266GB avail) - /var/drives/zeta/gh (266GB avail) - /var/drives/theta/gh (266GB avail) - /var/drives/lambda/gh (266GB avail) - /$
Sep 14 09:09:42 DEBUG write:   Saving 1 metadata files for Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:09:42 DEBUG write:     Saving metadata in /var/drives/alpha/gh/.gh_metastore/Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:09:42 DEBUG write:     Saving backup metadata file in /var/drives/theta/gh/.gh_metastore_backup/Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:09:42 DEBUG write:   Copying 2.00GB file to /var/drives/alpha/gh/Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:10:14 DEBUG write:     Copy created at 64.1 MBps.
Sep 14 09:10:14 DEBUG write:     Creating symlink in share pointing to the above file copy.
Sep 14 09:10:14 DEBUG write:   Saving 1 metadata files for Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:10:14 DEBUG write:     Saving metadata in /var/drives/alpha/gh/.gh_metastore/Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:10:14 DEBUG write:     Saving backup metadata file in /var/drives/theta/gh/.gh_metastore_backup/Movies/Racing Stripes (2005)/Racing Stripes (2005).mkv
Sep 14 09:10:16 DEBUG read_smb_spool: Processing Samba spool...
Sep 14 09:10:16 DEBUG read_smb_spool: Directory created: Movies/@^]Bw�U
Sep 14 09:10:17 DEBUG read_smb_spool: Found 11 new tasks in spool.
Sep 14 09:10:17 INFO write: Now working on task ID 581865: write Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
Sep 14 09:10:17 INFO write: File created: Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv - 58.8MB
Sep 14 09:10:17 DEBUG write: Loading metafiles for Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv ...
Sep 14 09:10:17 DEBUG write:   Got 0 metadata files.
Sep 14 09:10:17 DEBUG write: 0 metafiles loaded.
Sep 14 09:10:17 DEBUG write:   Drives with available space: /var/drives/zeta/gh (266GB avail) - /var/drives/theta/gh (266GB avail) - /var/drives/lambda/gh (266GB avail) - /var/drives/kappa/gh (266GB avail) - /$
Sep 14 09:10:17 DEBUG write:   Saving 1 metadata files for Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
Sep 14 09:10:17 DEBUG write:     Saving metadata in /var/drives/zeta/gh/.gh_metastore/Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
Sep 14 09:10:17 DEBUG write:     Saving backup metadata file in /var/drives/theta/gh/.gh_metastore_backup/Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
Sep 14 09:10:17 DEBUG write:   Copying 58.8MB file to /var/drives/zeta/gh/Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
Sep 14 09:10:18 DEBUG write:     Copy created at 58.8 MBps.
Sep 14 09:10:18 DEBUG write:     Creating symlink in share pointing to the above file copy.
Sep 14 09:10:18 DEBUG write:   Saving 1 metadata files for Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
Sep 14 09:10:18 DEBUG write:     Saving metadata in /var/drives/zeta/gh/.gh_metastore/Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
Sep 14 09:10:18 DEBUG write:     Saving backup metadata file in /var/drives/theta/gh/.gh_metastore_backup/Movies/Racing Stripes (2005)/Voice Acting-featurette.mkv
````

**Comment by gboudreau** [comment]
Something is creating those folders on your shares.
> Sep 14 09:10:16 DEBUG read_smb_spool: Directory created: Movies/@^]Bw�U

That indicates the Greyhole VFS module was notified that this folder was created on your share.

**Comment by Sparticuz** [comment]
So what should I do to troubleshoot that? All I did was cut and paste the files in Explorer.

**Comment by gboudreau** [comment]
Not sure. Maybe increasing the verbosity of the Samba logs would show you what client creates those folders, or maybe you need a Samba VFS that allows you to monitor changes on your shares..? But even if you find what remote client created those, that might not really help... Your Samba server can't know which process created those folders on the remote computer.

Googling `samba strange folder names` returns results for posts from ppl experiencing something similar. Maybe you can find the answer in one of those..?

**Comment by gboudreau** [comment]
This one seems like a good explanation:
https://superuser.com/questions/458995/files-folders-get-weird-names-and-become-inaccessible-on-samba-share

So is this one:
https://serverfault.com/questions/124611/special-characters-in-samba-filenames

**Comment by Sparticuz** [comment]
Sure, I'll look into those and report back. Thanks for helping rule out Gh

**Comment by Sparticuz** [comment]
I feel like there is some kind of character encoding problem.
````
kyle@haven 11:55:48 /mnt/samba/Backup $ rm -rf Kristyn/
kyle@haven 11:55:54 /mnt/samba/Backup $ ll
total 4.0K
drwxr-xr-x  2 kyle sambaread    0 Sep 14 11:55  ./
dr-xr-xr-x 15 kyle kyle      4.0K Sep 13 08:17  ../
drwxr-xr-x  2 kyle sambaread    0 Sep 13 12:02  CrashPlan/
drwxr-xr-x  2 kyle sambaread    0 Sep 14 11:45 "Kyle's Phone Pics"/
drwxr-xr-x  2 kyle sambaread    0 Sep 14 11:45 'New folder'/
drwxr-xr-x  2 kyle sambaread    0 Sep 14 11:45 'New MotoG Backup'/
drwxr-xr-x  2 kyle sambaread    0 Sep 13 16:26  Sort/
drwxr-xr-x  2 kyle sambaread    0 Sep 14 11:45 'The Great Disk Rip of 2016'/
drwxr-xr-x  2 kyle sambaread    0 Sep 14 11:45 'To Convert'/
drwxr-xr-x  2 kyle sambaread    0 Sep 14 11:45 'Wedding Raw'/
kyle@haven 11:55:59 /mnt/samba/Backup $
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
kyle@haven 11:55:30 /mnt/samba/Backup $ sudo greyhole --logs
[sudo] password for kyle:
Sep 14 11:55:34 DEBUG sleep: Nothing to do... Sleeping.
Sep 14 11:55:45 DEBUG sleep: Nothing to do... Sleeping.
Sep 14 11:55:55 DEBUG read_smb_spool: Processing Samba spool...
Sep 14 11:55:55 DEBUG read_smb_spool: Found 3 new tasks in spool.
Sep 14 11:55:55 INFO rmdir: Now working on task ID 583258: rmdir Backup/
Sep 14 11:55:55 INFO rmdir: Directory deleted: /var/shares/backup/
Sep 14 11:55:55 INFO rmdir: Now working on task ID 583259: rmdir Backup/V?<V
Sep 14 11:55:55 INFO rmdir: Directory deleted: /var/shares/backup/V?<V
Sep 14 11:55:55 INFO rmdir: Now working on task ID 583260: rmdir Backup/
Sep 14 11:55:55 INFO rmdir: Directory deleted: /var/shares/backup/
Sep 14 11:55:55 DEBUG sleep: Nothing to do... Sleeping.
Sep 14 11:56:05 DEBUG read_smb_spool: Processing Samba spool...
Sep 14 11:56:05 DEBUG sleep: Nothing to do... Sleeping.
Sep 14 11:56:15 DEBUG sleep: Nothing to do... Sleeping.
````

**Comment by Sparticuz** [comment]
Just an Update:

It looks like there is some kind of character encoding problem. I've set Samba to UTF-8, (which is the default, and should have already been). I've also set PHP to UTF-8. I'm having to go through all of my drives and shares, converting all the filenames to UTF-8, using convmv. 

After that, I'm going to have to go through all my drives, all my shares, and all my .gh_metastores, and delete the bad folders. Then hopefully after that, I can enable GH and run a fsck.

**Comment by Sparticuz** [comment]
I've cleaned out all the bad folders. Here's what I've got:
````
kyle@haven 01:22:37 /mnt/samba/Music $ touch test
--------------
Sep 15 13:22:46 DEBUG read_smb_spool: Found 1 new tasks in spool.
Sep 15 13:22:46 INFO write: Now working on task ID 584864: write Music/test
Sep 15 13:22:46 INFO write: File created: Music/test - 0.00B
Sep 15 13:22:46 DEBUG write: Loading metafiles for Music/test ...
Sep 15 13:22:46 DEBUG write:   Got 0 metadata files.
Sep 15 13:22:46 DEBUG write: 0 metafiles loaded.
Sep 15 13:22:46 DEBUG write:   Drives with available space: /var/drives/kappa/gh (610GB avail) - /var/drives/theta/gh (542GB avail) - /var/drives/lambda/gh (423GB avail) - /var/drives/epsilon/gh (384GB avail) - /var/drives/alpha/gh (373GB avail) - /var/drives/zeta/gh (370GB avail)
Sep 15 13:22:46 DEBUG write:   Saving 2 metadata files for Music/test
Sep 15 13:22:46 DEBUG write:     Saving metadata in /var/drives/theta/gh/.gh_metastore/Music/test
Sep 15 13:22:46 DEBUG write:     Saving metadata in /var/drives/kappa/gh/.gh_metastore/Music/test
Sep 15 13:22:46 DEBUG write:   Copying 0.00B file to /var/drives/theta/gh/Music/test
Sep 15 13:22:46 DEBUG write:     Creating symlink in share pointing to the above file copy.
Sep 15 13:22:46 DEBUG write:   Saving 2 metadata files for Music/test
Sep 15 13:22:46 DEBUG write:     Saving metadata in /var/drives/theta/gh/.gh_metastore/Music/test
Sep 15 13:22:46 DEBUG write:     Saving metadata in /var/drives/kappa/gh/.gh_metastore/Music/test
Sep 15 13:22:46 DEBUG write:   Copying 0.00B file to /var/drives/kappa/gh/Music/test
Sep 15 13:22:46 DEBUG write:   Saving 2 metadata files for Music/test
Sep 15 13:22:46 DEBUG write:     Saving metadata in /var/drives/theta/gh/.gh_metastore/Music/test
Sep 15 13:22:46 DEBUG write:     Saving metadata in /var/drives/kappa/gh/.gh_metastore/Music/test
Sep 15 13:22:46 DEBUG sleep: Nothing to do... Sleeping.
-------------
kyle@haven 01:22:39 /mnt/samba/Music $ mkdir testfolder
-------------
Sep 15 13:23:07 DEBUG read_smb_spool: Processing Samba spool...
Sep 15 13:23:07 DEBUG read_smb_spool: Directory created: Music/ $BU
Sep 15 13:23:07 DEBUG sleep: Nothing to do... Sleeping.
````
testfolder, get's created: /var/shares/music/testfolder

BUT:
````
kyle@haven 01:27:39 /var/drives/epsilon/gh/Music $ ll
total 28K
drwxrwxr-x  3 kyle kyle  20K Sep 15 13:23 ./
drwxrwxr-x 16 kyle kyle 4.0K Sep 15 13:04 ../
drwxrwxrwx  2 root root 4.0K Sep 15 13:23 \ $B\237\242U/

kyle@haven 01:28:46 /var/drives/theta/gh/Music $ ll
total 28K
drwxrwxr-x  3 kyle kyle  20K Sep 15 13:23 ./
drwxrwxr-x 16 kyle kyle 4.0K Sep 15 13:04 ../
drwxrwxrwx  2 root root 4.0K Sep 15 13:23 \ $B\237\242U/
````
So because the files in the LZ got created properly, I have a feeling it's GH related. It looks like mkdir isn't logged into the tasks_completed table in sql, otherwise, I'd give you what was there also.

**Comment by gboudreau** [comment]
Greyhole creates the `\ $B\237\242U/` folders in the data drives because it was told by Samba that this folder was created on the share. `mkdir` operations are not logged in MySQL; they are processed as soon as the Greyhole daemon is notified by the Greyhole VFS module, thus why it's logged with the `read_smb_spool` tag.
You can try to shut down the greyhole daemon, and mkdir that folder, and look in `/var/spool/greyhole`. The files in there are simple text files that contain what Samba told the Greyhole's VFS module happened on your Greyhole shares.

**Comment by Sparticuz** [comment]
````
kyle@haven 02:37:08 /var/spool/greyhole $ cat 1505500626395871
mkdir
Music
AU
````

**Comment by Sparticuz** [comment]
I'm guessing it should say testfolder2 instead of AU?

**Comment by gboudreau** [comment]
Yes. That string is what samba sent to the Greyhole VFS as the `const char *path` parameter of the `mkdir` function.
Ref: https://github.com/gboudreau/Greyhole/blob/master/samba-module/vfs_greyhole-samba-4.x.c#L123

**Comment by Sparticuz** [comment]
I'm taking a look at my vfs module now. It looks like mine is 52k, while yours is 44k. I might have never overwritten my compiled one with yours. I'll do that now.

**Comment by Sparticuz** [comment]
I'm out of ideas
````
kyle@haven 02:37:06 /mnt/samba/Music $ mkdir testfolder3
kyle@haven 02:53:35 /var/spool/greyhole $ cat 1505501612547750
mkdir
Music
U

````

**Comment by gboudreau** [comment]
`hexdump -C 1505501612547750` ?

**Comment by Sparticuz** [comment]
kyle@haven 02:54:58 /var/spool/greyhole $ hexdump -c 1505501612547750
0000000   m   k   d   i   r  \n   M   u   s   i   c  \n 200 265 241 222
0000010 204   U  \n  \n
0000014

**Comment by gboudreau** [comment]
Capital `C`

**Comment by Sparticuz** [comment]
lol, sorry
````
kyle@haven 02:58:13 /var/spool/greyhole $ hexdump -C 1505501612547750
00000000  6d 6b 64 69 72 0a 4d 75  73 69 63 0a 80 b5 a1 92  |mkdir.Music.....|
00000010  84 55 0a 0a                                       |.U..|
00000014
````

**Comment by gboudreau** [comment]
So the VFS module receive the string `80 b5 a1 92 84 55`...
Can you repeat with slightly different folder names, and post the hexdump of each. Maybe we can find a pattern in the folder name you use, and what the VFS module receives..?

**Comment by gboudreau** [comment]
What version of Samba are you using?

**Comment by Sparticuz** [comment]
I'm on Arch, latest samba 4.6.7

````
kyle@haven 02:53:32 /mnt/samba/Music $ mkdir testfolder4
kyle@haven 02:59:08 /var/spool/greyhole $ hexdump -C 1505502111804982
00000000  6d 6b 64 69 72 0a 4d 75  73 69 63 0a 80 b5 a1 92  |mkdir.Music.....|
00000010  84 55 0a 0a                                       |.U..|
00000014
````
````
kyle@haven 03:01:51 /mnt/samba/Music $ mkdir testfolder5
kyle@haven 03:03:34 /var/spool/greyhole $ hexdump -C 1505502205835516
00000000  6d 6b 64 69 72 0a 4d 75  73 69 63 0a 80 b5 a1 92  |mkdir.Music.....|
00000010  84 55 0a 0a                                       |.U..|
00000014
````
Maybe I'm missing something, but these all look the same?

**Comment by gboudreau** [comment]
Something else to try: enable the Samba-provided `audit` VFS module, to see what it logs.
It should log to syslog if you change the Music share definition in smb.conf to:
`vfs objects = audit greyhole` instead of just `vfs objects = greyhole`

**Comment by Sparticuz** [comment]
````
Sep 15 15:06:12 haven smbd_audit[2167]: connect to service Music by user kyle
Sep 15 15:06:12 haven smbd_audit[2167]: chmod_acl testfolder6 mode 0x1fd failed: No data available
Sep 15 15:06:12 haven smbd_audit[2167]: mkdir testfolder6
````

**Comment by gboudreau** [comment]
Looks like samba changed the VFS methods definitions in 4.6 vs 4.5:
https://github.com/samba-team/samba/blob/67466271c26ca10c6bce6a51f9dd97a22258f8a4/source3/include/vfs.h#L173
2nd parameter was a `const char *` in 4.5, and it was changed to a `struct smb_filename *` in 4.6...

Did your Samba upgrade from 4.5 to 4.6 recently (when this problem started)?

**Comment by Sparticuz** [comment]
That might have been when it started. I do remember having difficulty with GH when I did, but I think reinstalling GH solved it. I think I posted about it. If that changed, I can take a look at the vfs module and see if I can fix it. I'm not an experienced C programmer, but I can take a crack at it.

On another note, is there a reason why GH doesn't use vfs_audit or vfs_full_audit instead of rolling vfs_greyhole? That might solve this problem, as well as #134

**Comment by gboudreau** [comment]
Those modules send logs to syslog/log files; Greyhole would need to parse those log files to get what it needs, if it can find everything it needs in them. I doubt that's a good idea.

**Comment by gboudreau** [comment]
I modified the VFS module for 4.6 in a branch: https://github.com/gboudreau/Greyhole/tree/4.6-mkdir-rmdir-issue
Trying to compile it now.

**Comment by Sparticuz** [comment]
Great! Well, either way, it looks like vfs_greyhole is modeled after vfs_audit. Since I'm usually one of the first to upgrade (because I'm on Arch), I'll make sure to start looking over vfs_audit before creating new patches.

**Comment by gboudreau** [comment]
What happens if you do this on your share?
```
mkdir testfolder7
echo 1 > testfolder7/testfile
```
Does greyhole correctly create the `testfolder7` folder, to put `testfile` in it, after failing to mkdir the proper folder when it processed the mkdir?

**Comment by Sparticuz** [comment]
I think it works. I never had any problems with files or subfolders, It just created screwy folders on the top level.
````
kyle@haven 03:41:14 /var/spool/greyhole $ ll
total 20K
drwxrwxrwx 3 root root 4.0K Sep 15 15:41 ./
drwxr-xr-x 5 root root 4.0K Sep 15 11:34 ../
drwxrwxrwt 2 root root   60 Sep 15 15:41 mem/
-rw-rw-rw- 1 kyle kyle   20 Sep 15 15:41 1505504462461300
-rw-rw-rw- 1 kyle kyle   47 Sep 15 15:41 1505504471291294
-rw-rw-rw- 1 kyle kyle   15 Sep 15 15:41 1505504471292221
kyle@haven 03:41:15 /var/spool/greyhole $ cat *
mkdir
Music
DU

open
Music
testfolder7/testfile
9
for writing
close
Music
9

````

**Comment by gboudreau** [comment]
Try this VFS module:
[greyhole-x86_64.so.zip](https://github.com/gboudreau/Greyhole/files/1307629/greyhole-x86_64.so.zip)

**Comment by Sparticuz** [comment]
ok, I put it in /usr/lib64/greyhole/greyhole-samba46.so
starting smbd....
````
kyle@haven 03:58:33 /mnt/samba/Music $ mkdir testfolder8
kyle@haven 03:58:49 /mnt/samba/Music $ touch testfolder8/testfile
-----------
kyle@haven 03:58:38 /var/spool/greyhole $ ll
total 20K
drwxrwxrwx 3 root root 4.0K Sep 15 15:58 ./
drwxr-xr-x 5 root root 4.0K Sep 15 11:34 ../
drwxrwxrwt 2 root root   40 Sep 15 15:58 mem/
-rw-rw-rw- 1 kyle kyle   25 Sep 15 15:58 1505505529263355
-rw-rw-rw- 1 kyle kyle   48 Sep 15 15:58 1505505538365499
-rw-rw-rw- 1 kyle kyle   16 Sep 15 15:58 1505505538366163
kyle@haven 03:59:01 /var/spool/greyhole $ cat *
mkdir
Music
testfolder8

open
Music
testfolder8/testfile
31
for writing
close
Music
31

````
Seems to be working! Let me start greyhole
````
Sep 15 16:00:27 INFO daemon: Greyhole (version 0.10.12) daemon started.
Sep 15 16:00:27 INFO daemon: Optimizing MySQL tables...
Sep 15 16:00:27 DEBUG daemon: Loading metadata store backup directories...
Sep 15 16:00:27 DEBUG daemon:   Found 2 directories in the settings table.
Sep 15 16:00:28 DEBUG daemon: Checking symlink at /usr/lib/samba/vfs/greyhole.so...
Sep 15 16:00:28 DEBUG daemon:   Is OK.
Sep 15 16:00:28 DEBUG read_smb_spool: Processing Samba spool...
Sep 15 16:00:28 DEBUG read_smb_spool: Directory created: Music/testfolder8
Sep 15 16:00:28 DEBUG read_smb_spool: Found 1 new tasks in spool.
Sep 15 16:00:28 INFO write: Now working on task ID 584865: write Music/testfolder8/testfile
Sep 15 16:00:28 INFO write: File created: Music/testfolder8/testfile - 0.00B
Sep 15 16:00:28 DEBUG write: Loading metafiles for Music/testfolder8/testfile ...
Sep 15 16:00:28 DEBUG write:   Got 0 metadata files.
Sep 15 16:00:28 DEBUG write: 0 metafiles loaded.
Sep 15 16:00:28 DEBUG write:   Drives with available space: /var/drives/kappa/gh (610GB avail) - /var/drives/theta/gh (542GB avail) - /var/drives/lambda/gh (423GB avail) - /var/drives/epsilon/gh (384GB avail) - /var/drives/alpha/gh (373GB avail) - /var/drives/zeta/gh (370GB avail)
Sep 15 16:00:28 DEBUG write:   Saving 2 metadata files for Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:     Saving metadata in /var/drives/theta/gh/.gh_metastore/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:     Saving metadata in /var/drives/kappa/gh/.gh_metastore/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:   Copying 0.00B file to /var/drives/theta/gh/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:     Creating symlink in share pointing to the above file copy.
Sep 15 16:00:28 DEBUG write:   Saving 2 metadata files for Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:     Saving metadata in /var/drives/theta/gh/.gh_metastore/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:     Saving metadata in /var/drives/kappa/gh/.gh_metastore/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:   Copying 0.00B file to /var/drives/kappa/gh/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:   Saving 2 metadata files for Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:     Saving metadata in /var/drives/theta/gh/.gh_metastore/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG write:     Saving metadata in /var/drives/kappa/gh/.gh_metastore/Music/testfolder8/testfile
Sep 15 16:00:28 DEBUG sleep: Nothing to do... Sleeping.
````
I think that did it!

**Comment by gboudreau** [comment]
W0rd! Will commit that fix, and build the VFS modules for all architectures using that.
Thanks for persevering! :)

**Comment by ghrasko** [comment]
Hi,

I just installed Fedora 25 with Amahi 10. From the Amahi Dashboard I installed the Greyhole UI and I am experiencing a similar problem. 

The odd filenames appear when I create an empty dirextory in Samba shares (mkdir routines in Greyhole). If I copy a folder with files from my Windows PC to the shared folder (Greyhole ZL), the directory is created correctly as it is not using the mkdir routins this case. However for my installation if I delete a folder with the LZ, the empty folders are not deleted from the Greyhole pool(s). This is not obvious for a normal Samba user as the shared directory is fine. But if you check the pool drive, the empty folders are there for ever.

I also checked the Samba and Greyhole codes and realized that the mkdir and rmdir event routiness parameter had been changed and my assumption was that there is something uninitialized structure field was retrieved. However my other guess was also some encoding issue.

The description with logs and spool files can be senn here. I don't know if this is the same issue or another one:

https://forums.amahi.org/viewtopic.php?f=4&t=7077

Gábor
